src/main/java/com/dev/cinema/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,37 @@
+package com.dev.cinema.security;
+
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import java.util.Optional;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> optionalUser = userService.findByEmail(email);
```

Comment:
let's use orElseThrow here instead, this will make code shorter

Reply:
done